hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -520,12 +520,20 @@ public Object get(Data key, boolean backup, Address
         return value;
     }
 
+    /**
+     * This method is called directly by user threads, in other words
+     * it is called outside of the partition threads.
+     */
     @Override
     public Data readBackupData(Data key) {
         Record record = getRecord(key);
 
         if (record == null) {
             return null;
+        } else {
+            if (partitionService.isPartitionOwner(partitionId)) {
+                accessRecord(record, Clock.currentTimeMillis());
```

Comment:
It doesn't look like `accessRecord` works well under the concurrent access, there is a lot of non-volatile and non-atomic operations inside it. I assume it's an intentional compromise and that's good enough for stats, right?

Reply:
Good point. Amended it by only updating last access time.